src/utils/config.ts

```diff
@@ -755,11 +762,38 @@ export const overrideEnvVariables = async (targetName: string) => {
   }
 }
 
-const getPrecedenceOfInsecureRequests = (
+const getPrecedenceOfHttpsAgentOptionsAndContent = async (
   config: Configuration,
   target: TargetJson
-): boolean => {
-  return target.allowInsecureRequests ?? !!config.allowInsecureRequests
+): Promise<HttpsAgentOptions> => {
+  const httpsAgentOptions = getHttpsAgentOptionsForInsecure({
+    allowInsecureRequests: false,
+    ...config.httpsAgentOptions,
+    ...target.httpsAgentOptions
+  })
+
+  if (httpsAgentOptions.caPath) {
+    httpsAgentOptions.ca = await readFile(httpsAgentOptions.caPath)
+  }
+  if (httpsAgentOptions.keyPath) {
+    httpsAgentOptions.ca = await readFile(httpsAgentOptions.keyPath)
+  }
+  if (httpsAgentOptions.certPath) {
+    httpsAgentOptions.cert = await readFile(httpsAgentOptions.certPath)
+  }
+
+  return httpsAgentOptions
+}
```

Comment:
While extracting `Target`:

Extracting contents of `caPath`, `certPath`, `keyPath` and pass to `ca`, `cert`, `key` respectively.